### ISSUE TITLE
Feature request: proper MSC3823 support in Admin API

### ISSUE THREAD

**Issue by aine-etke** [element-hq/synapse#18016]
**Description:**

Hello,

I'm working on https://github.com/etkecc/synapse-admin/pull/195 to support the MSC3823 feature in Synapse Admin, and currently (at the moment of writing using the `develop` Synapse docker tag) the following things are missing:
* Support `suspended` as one of the [List Accounts](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#list-accounts) params and response keys
* ~~Support `suspended` as one of the [Query User Account](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#query-user-account) response keys~~ Done
* Support `suspended` as one of the [Create or Modify Account](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#create-or-modify-account) body params

[MSC3823 is already considered stable](https://github.com/element-hq/synapse/pull/17964), but it's almost unusable in Admin API without those changes.

Thank you in advance!

PS: initially the clarifications were asked in https://github.com/element-hq/synapse/pull/17255#issuecomment-2525105226

**Comment by aine-etke** [comment]
cc @element-hq/trust-safety

**Comment by anoadragon453** [comment]
We found out from [this comment](https://github.com/element-hq/synapse/pull/17255#issuecomment-2528807517) that a `suspended` field is already returned by the [Query User Account](https://element-hq.github.io/synapse/develop/admin_api/user_admin_api.html#query-user-account) Admin API (only visible in the `develop` version of the docs currently).

@aine-etke from your issue description, do I understand correctly that you'd still like a `suspended` field to be added to the [List Accounts](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#list-accounts) and [Create or Modify Account](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#create-or-modify-account) Admin APIs?

It turns out that there's a [currently-undocumented Admin API](https://github.com/element-hq/synapse/blob/adce8a0111a37795bc4167afb2543a8877320b6a/synapse/rest/admin/users.py#L739-L766) for suspending accounts. Though now I'm thinking that it indeed makes more sense to just fold this into Create or Modify account...

**Comment by aine-etke** [comment]
> do I understand correctly that you'd still like a suspended field to be added to the [List Accounts](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#list-accounts) and [Create or Modify Account](https://element-hq.github.io/synapse/latest/admin_api/user_admin_api.html#create-or-modify-account) Admin APIs?

Correct. I'd expect it to be the same way as locked and/or deactivated

> It turns out that there's a [currently-undocumented Admin API](https://github.com/element-hq/synapse/blob/adce8a0111a37795bc4167afb2543a8877320b6a/synapse/rest/admin/users.py#L739-L766) for suspending accounts. Though now I'm thinking that it indeed makes more sense to just fold this into Create or Modify account...

Correct.  Again, I'd expect this new feature to use the same pattern as locked and/or deactivated, i.e., presented in the List Accounts, Query User Account endpoints, plus could be set/unset using the Create or Modify Account endpoint

**Comment by anoadragon453** [comment]
@aine-etke would you be upset if that undocumented suspend API went away? :stuck_out_tongue:

**Comment by turt2live** [comment]
> would you be upset if that undocumented suspend API went away? 😛

Matrix.org's Trust & Safety team would be mildly upset if it disappeared. The intention is to document it rather than remove it.

**Comment by anoadragon453** [comment]
@turt2live I was curious whether it was in use by anyone. Thanks for letting us know!
